main.tf

```diff
@@ -0,0 +1,102 @@
+provider "alicloud" {
+  profile                 = var.profile != "" ? var.profile : null
+  shared_credentials_file = var.shared_credentials_file != "" ? var.shared_credentials_file : null
+  region                  = var.region != "" ? var.region : null
+  skip_region_validation  = var.skip_region_validation
+  configuration_source    = "terraform-alicloud-modules/ecs-image"
+}
+
+#####################
+# create image by instance_id
+#####################
+resource "alicloud_image" "instance" {
+  count = length(var.instance_id) > 0 ? length(var.instance_id) : 0
+
+  architecture      = var.image_create_architecture
+  instance_id       = var.instance_id[count.index]
+  description       = var.description
+  platform          = var.image_create_platform
+  resource_group_id = var.resource_group_id
+  force             = var.force
+  tags = merge({
+    create = "TF-instance_id"
+  }, var.tags)
+}
+
+#####################
+# create image by snapshot_id
+#####################
+resource "alicloud_image" "snapshot" {
+  count = length(var.snapshot_id) > 0 ? length(var.snapshot_id) : 0
+
+  architecture      = var.image_create_architecture
+  snapshot_id       = var.snapshot_id[count.index]
+  description       = var.description
```

Comment:
description       = "An image comes from snapshot var.snapshot_id and created by Terraform."